pkg/session/session.go

```diff
@@ -2351,23 +2351,6 @@ func runStmt(ctx context.Context, se *session, s sqlexec.Statement) (rs sqlexec.
 
 	sessVars := se.sessionVars
 
-	// Record diagnostic information for DML statements
```

Comment:
This seems to be a behavior change. Before this PR, `LastQueryInfo` is record even if this function returns error, but after this change, if `runStmt` returns error and the returned RecordSet is nil, `FinishExecuteStmt` won't be called so LastQueryInfo won't be recorded either.

Reply:
Most errors occur during the execution stage, I think it's reasonable move to here.  and you can see the manual test on the issue, I kill the pd, it can record `pd timeout` error during execution.

Reply:
cc @djshow832 is it ok to move to FinishExecuteStmt?